src/opam.ml

```diff
@@ -288,8 +299,11 @@ let remove t switch packages =
 let switch_compiler t switch =
   let open Promise.Syntax in
   let+ switch_state = Switch_state.of_switch t switch in
-  let compilers = Switch_state.compilers switch_state in
-  Option.bind compilers ~f:List.hd
+  match switch_state with
+  | Some switch_state ->
+    let compilers = Switch_state.compilers switch_state in
+    Option.bind compilers ~f:List.hd
+  | None -> None
```

Comment:
This could look like 

```ocaml
let switch_compiler t switch =
  let open Promise.Syntax in
  let+ switch_state = Switch_state.of_switch t switch in
  let open Option.O in
  let* switch_state (* this is called "let-punning" *)
  in
  let* compilers = Switch_state.compilers switch_state in
  List.hd compilers
```

Reply:
> (* this is called "let-punning" *)
Thank you for the info. I wasn't sure to understand correctly what it was. I have to take a deeper look at the doc of Ocaml.